src/network.c

```diff
@@ -585,22 +555,22 @@ int recv_any_msg(Connection* conn, int* type, void* msg, int* len,
  * @param socketfd The socket
  * @param buf The data
  * @param amount The data size
- * @return The number of bytes written, 0 on fatal error, and -1 on recoverable
- *         error.
+ * @return The number of bytes written, -1 on fatal error, and 0 on recoverable
+ *         error, just like write().
  */
 int write_raw(int socketfd, const char* buf, int amount) {
   int n;
   n = write(socketfd, buf, amount);
-  if (n == -1) {
+  if (n == 0) {
```

Comment:
The verbage about what happens when count is zero caused me to get distracted. You are right: only -1 is an error.